src/main/host/thread_preload.c

```diff
@@ -499,34 +496,33 @@ long threadpreload_nativeSyscall(Thread* base, long n, va_list args) {
 }
 
 Thread* threadpreload_new(Host* host, Process* process, gint threadID) {
-    ThreadPreload* thread = g_new0(ThreadPreload, 1);
-
-    thread->base = (Thread){.run = threadpreload_run,
-                            .resume = threadpreload_resume,
-                            .terminate = threadpreload_terminate,
-                            .getReturnCode = threadpreload_getReturnCode,
-                            .isRunning = threadpreload_isRunning,
-                            .free = threadpreload_free,
-                            .newClonedPtr = threadpreload_newClonedPtr,
-                            .releaseClonedPtr = threadpreload_releaseClonedPtr,
-                            .getReadablePtr = threadpreload_getReadablePtr,
-                            .getReadableString = threadpreload_getReadableString,
-                            .getWriteablePtr = threadpreload_getWriteablePtr,
-                            .flushPtrs = threadpreload_flushPtrs,
-                            .nativeSyscall = threadpreload_nativeSyscall,
-                            .type_id = THREADPRELOAD_TYPE_ID,
-                            .referenceCount = 1};
-    MAGIC_INIT(&thread->base);
-
-    thread->threadID = threadID;
-    thread->sys =
-        syscallhandler_new(host, process, _threadPreloadToThread(thread));
+    ThreadPreload* thread = g_new(ThreadPreload, 1);
+
+    *thread = (ThreadPreload){
+        .base = thread_create(host, process, THREADPRELOAD_TYPE_ID,
+                              (ThreadMethods){
+                                  .run = threadpreload_run,
+                                  .resume = threadpreload_resume,
+                                  .terminate = threadpreload_terminate,
+                                  .getReturnCode = threadpreload_getReturnCode,
+                                  .isRunning = threadpreload_isRunning,
+                                  .free = threadpreload_free,
+                                  .newClonedPtr = threadpreload_newClonedPtr,
+                                  .releaseClonedPtr = threadpreload_releaseClonedPtr,
+                                  .getReadablePtr = threadpreload_getReadablePtr,
+                                  .getReadableString = threadpreload_getReadableString,
+                                  .getWriteablePtr = threadpreload_getWriteablePtr,
+                                  .flushPtrs = threadpreload_flushPtrs,
+                                  .nativeSyscall = threadpreload_nativeSyscall,
+                              }),
+        .ptr_to_block = g_hash_table_new(g_int64_hash, g_int64_equal),
```

Comment:
The indentation of the ThreadMethods seems odd. Did you run clang-format yet?

Reply:
I hadn't, but doing so didn't change this. It just ends up getting indented more due the extra level of nesting here (initializing the struct that we pass to `thread_create`)